pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -39,210 +40,29 @@ var gatheringFunctionBuilders = map[GatheringFunctionName]GathererFunctionBuilde
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "KubePodNotReady",
-				TailLines: 100,
-				Previous:  false,
-			},
-			GatherPodDefinition: GatherPodDefinitionParams{
-				AlertName: "KubePodNotReady",
-			},
-		},
-	},
-	{
-		Conditions: []ConditionWithParams{
-			{
-				Type: AlertIsFiring,
-				Alert: &AlertConditionParams{
-					Name: "AlertmanagerFailedToSendAlerts",
-				},
-			},
-		},
-		GatheringFunctions: GatheringFunctions{
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "AlertmanagerFailedToSendAlerts",
-				Container: "alertmanager",
-				TailLines: 50,
-			},
-		},
-	},
-	{
-		Conditions: []ConditionWithParams{
-			{
-				Type: AlertIsFiring,
-				Alert: &AlertConditionParams{
-					Name: "PrometheusOperatorSyncFailed",
-				},
-			},
-		},
-		GatheringFunctions: GatheringFunctions{
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "PrometheusOperatorSyncFailed",
-				Container: "prometheus-operator",
-				TailLines: 50,
-			},
-		},
-	},
-	{
-		Conditions: []ConditionWithParams{
-			{
-				Type: AlertIsFiring,
-				Alert: &AlertConditionParams{
-					Name: "AlertmanagerFailedReload",
-				},
-			},
-		},
-		GatheringFunctions: GatheringFunctions{
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "AlertmanagerFailedReload",
-				Container: "alertmanager",
-				TailLines: 50,
-			},
-		},
-	},
-	{
-		Conditions: []ConditionWithParams{
-			{
-				Type: AlertIsFiring,
-				Alert: &AlertConditionParams{
-					Name: "PrometheusTargetSyncFailure",
-				},
-			},
-		},
-		GatheringFunctions: GatheringFunctions{
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "PrometheusTargetSyncFailure",
-				Container: "prometheus",
-				TailLines: 50,
-			},
-		},
-	},
-	{
-		Conditions: []ConditionWithParams{
-			{
-				Type: AlertIsFiring,
-				Alert: &AlertConditionParams{
-					Name: "ThanosRuleQueueIsDroppingAlerts",
-				},
-			},
-		},
-		GatheringFunctions: GatheringFunctions{
-			GatherContainersLogs: GatherContainersLogsParams{
-				AlertName: "ThanosRuleQueueIsDroppingAlerts",
-				Container: "thanos-ruler",
-				TailLines: 50,
-			},
-		},
-	},
-}
-
 // Gatherer implements the conditional gatherer
 type Gatherer struct {
 	gatherProtoKubeConfig   *rest.Config
 	metricsGatherKubeConfig *rest.Config
 	imageKubeConfig         *rest.Config
 	gatherKubeConfig        *rest.Config
 	// there can be multiple instances of the same alert
-	firingAlerts   map[string][]AlertLabels
-	gatheringRules []GatheringRule
-	clusterVersion string
+	firingAlerts                map[string][]AlertLabels
+	gatheringRules              []GatheringRule
+	clusterVersion              string
+	configurator                configobserver.Configurator
+	gatheringRulesServiceClient GatheringRulesServiceClient
+}
+
+type GatheringRulesServiceClient interface {
+	RecvGatheringRules(ctx context.Context, endpoint string) ([]byte, error)
 }
```

Comment:
Well, it would be harder to unit test it. Now I'm using a very clean and simple implementation of MockClient https://github.com/openshift/insights-operator/pull/583/files#diff-5819ed4e968701d806c6aac65c11f9e554f0924d6acb9f5de8a09f9e2bedbc8fR248

Reply:
Ahh OK, I haven't seen the tests yet. Then it's completely OK. Thanks